### PR TITLE
Adds building of `examples/todoApp` to CI

### PR DESCRIPTION
**Pull request by shayneczyzewski**
### Description

Adds building of `examples/todoApp` to CI. This should help catch any TS errors.

Example of a build failure: https://github.com/wasp-lang/wasp/actions/runs/4734719184/jobs/8404025469 (this was what would have happened on `main` before the fix was made).

### Select what type of change this PR introduces:

1. [x] **Just code/docs improvement** (no functional change). 
2. [ ] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### Update Waspc ChangeLog and version if needed
If you did a bug fix, new feature, or breaking change, that affects waspc, make sure you satisfy the following:
1. [ ] I updated [ChangeLog.md](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
2. [ ] I bumped waspc version in [waspc.cabal](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.

### CODE REVIEW

File: waspc/tools/ensure_todoapp_builds.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+echo "Running ensure_todoapp_builds.sh from $dir"
+
+cd "$dir/../examples/todoApp"
+echo "cd into $(pwd)"
+
+# Compile example app.
+WASP_BINARY_PATH="$(cabal list-bin wasp-cli)"
+CABAL_PROJECT_ROOT_PATH="$(cabal list-bin wasp-cli | sed s/\\/dist-newstyle.*//)"
+WASP_COMMAND="waspc_datadir=$CABAL_PROJECT_ROOT_PATH/data $WASP_BINARY_PATH"
+echo "Wasp command: $WASP_COMMAND"
```
Comment (Martinsos) [inline waspc/tools/ensure_todoapp_builds.sh:16]: Have you tried just running `cabal run wasp-cli build`? For me that works to build the waspc/examples/todoApp.

File: waspc/tools/ensure_todoapp_builds.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh -e
+
+# Compiles the todoApp and checks `npm run build` on both client and server.
```
Comment (Martinsos) [inline waspc/tools/ensure_todoapp_builds.sh:3]: Maybe provide a bit of context of why we care about this. You can explain that this is what is done during deploymen and that we want to make sure we have a way to test it without doign actual deployment, or smth like that?

File: waspc/tools/ensure_todoapp_builds.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh -e
```
Comment (Martinsos) [inline waspc/tools/ensure_todoapp_builds.sh:1]: You put this in separate file so we can also run it locally?

**Review by Martinsos — APPROVED** [review]

**Comment by Martinsos** [comment]
Great, this will help!!!

File: waspc/tools/ensure_todoapp_builds.sh (hunk 1 above)
Comment (shayneczyzewski) [reply waspc/tools/ensure_todoapp_builds.sh:16]: Ah yeah, that may work too, I'll try :D

File: waspc/tools/ensure_todoapp_builds.sh (hunk 3 above)
Comment (shayneczyzewski) [reply waspc/tools/ensure_todoapp_builds.sh:1]: We could, but just easier to manage this way mainly.

**Comment by Martinsos** [comment]
A thought here: this building step (running `npm build` in both projects, client and server), when does it happen? The only other place we do it at is Dockerfile, isn't it?

So what we really have is `wasp build` generating a code in .wasp/build, and then that code, we want to go there in those projects and build them, because that is what we finally want to deploy.

I am wondering if we should in some way extract this "building step" then, make it a standalone thing. Because in this PR we are really repeating logic that we have in that Dockerfile, I think? The idea of "building" the stuff in .wasp/build .

So maybe we should recognize that, and have a special thing for doing that, and then we can test that in CI + call it in Dockerfile + maybe Wasp users can call if they want, whatever.

I am not sure how it would be capture though. Maybe in .sh script at the top level of .wasp/build? If only `npm run build` go into it, maybe that is redundant then? What about prisma generation, I saw that step also in Dockerfile, should that be part of this "build" step?

**Pull request merged**